db/system_keyspace.cc

```diff
@@ -1944,6 +1944,116 @@ class describe_ring_table : public streaming_virtual_table {
     }
 };
 
+class snapshots_table : public streaming_virtual_table {
```

Comment:
Actually I'd prefer to do this as a follow-up. We have bunch of virtual table related patches in-flight. Doing this would cause great turmoil. Let's wait until the dust settle and then I will move them all at once.
Also there is another conflicting proposal from Avi/Pavel to introduce a virtual table registration API and let subsystems register their virtual tables. So yet another reason to wait until the dust settles.